File: src/1277-Count-Square-Submatrices-with-All-Ones/1277.java
class Solution {
    public int countSquares(int[][] A) {
        int res = 0;
        for (int i = 0; i < A.length; i++) {
            for (int j = 0; j < A[0].length; j++) {
                if (i > 0 && j > 0 && A[i][j] > 0)
                    A[i][j] = Math.min(A[i - 1][j], Math.min(A[i][j - 1], A[i - 1][j - 1])) + 1;
                res += A[i][j];
            }
        }
        return res;
    }
}

File: src/1277-Count-Square-Submatrices-with-All-Ones/1277.cpp
class Solution 
{
public:
    int countSquares(vector<vector<int>>& A) 
    {
        int res = 0;
        for (int i = 0; i < A.size(); i++) 
        {
            for (int j = 0; j < A[0].size(); j++)
            {
                if (i && j && A[i][j])
                    A[i][j] = min({A[i - 1][j], A[i][j - 1], A[i - 1][j - 1]}) + 1;
                res += A[i][j];
            }
        }
        return res;
    }
};